feat: AB#26095 add pdf meta service

maxrem opened the pull request:
- add gdpr meta ban list to config
- check for meta data in pdf doc info
- check for meta data in pdf /Metadata
- check for meta data in pdf images exif
- check for meta data in pdf images info
- add report to endpoint response
- add flag to endpoint to possibly ignore report

maxrem committed:
add pdf meta service

- add gdpr meta ban list to config
- check for meta data in pdf doc info
- check for meta data in pdf /Metadata
- check for meta data in pdf images exif
- check for meta data in pdf images info
- add report to endpoint response
- add flag to endpoint to possibly ignore report

JordyMoos commented on app/api/domains/publications/endpoints/publications/versions/attachments/upload_attachment_endpoint.py:
```diff
@@ -52,19 +55,29 @@ def post_upload_attachment_endpoint(
     file_binary = uploaded_file.file.read()
+    pdf_meta_report = pdf_meta_service.report_banned_meta(file_binary)
+
+    if not ignore_report and len(pdf_meta_report) > 0:
```
raise HTTPException(434, detail=pdf_meta_report)

maxrem committed:
move pdf meta service to 'other' domain

- add pdf meta service to file upload endpoint
- move FileData and use it for publication as well
- make enum variables uppercase
- add all file functionality to FileData
- add HTTPException to report
- remove main config ban list

maxrem committed:
change logic to not always make meta report if not necessary, make pip-upgrade
